src/server/passport.js

```diff
@@ -17,8 +16,11 @@ const configurePassport = db => {
     cb(null, user._id);
   });
   passport.deserializeUser((id, cb) => {
-    users.findOne({ _id: id }).then(user => {
-      cb(null, user);
+    users.findOne({_id: id }).then(user => {
+      if(user.provider != "ADFS"){
+        cb(null, false);
```

Comment:
WRONG!
provider can be different than ADFS. 
it can be user-service and is allowed.